pkg/controllers/management/clusterprovisioner/provisioner.go

```diff
@@ -488,10 +491,24 @@ func (p *Provisioner) reconcileCluster(cluster *v3.Cluster, create bool) (*v3.Cl
 		}
 	} else if spec.RancherKubernetesEngineConfig != nil && spec.RancherKubernetesEngineConfig.Restore.Restore {
 		logrus.Infof("Restoring cluster [%s] from backup", cluster.Name)
+		// cluster may need to be restored if key rotation fails
+		// ensure restore does not get short-circuited by key rotation since RKE checks for key rotation before restore
+		spec.RancherKubernetesEngineConfig.RotateEncryptionKey = false
 		apiEndpoint, serviceAccountToken, caCert, err = p.restoreClusterBackup(cluster, *spec)
+	} else if strings.Contains(apimgmtv3.ClusterConditionUpdated.GetMessage(cluster), errKeyRotationFailed.Error()) {
+		// prevent driver updates if the cluster needs to be restored
```

Comment:
Does this need a log line so we know it was returned as is with no error?

Reply:
Good point, I'll add some logging here